reports/src/main/java/com/ivy/reports/ReportViewModel.kt

```diff
@@ -1,415 +1,418 @@
 
-                false
-            }
-            .filter {
-                //Filter by Excluded Keywords
-
-                val excludedKeywords = filter.excludeKeywords
-                if (excludedKeywords.isEmpty()) return@filter true
-
-                if (it.title != null && it.title!!.isNotEmpty()) {
-                    excludedKeywords.forEach { keyword ->
-                        if (it.title!!.containsLowercase(keyword)) {
-                            return@filter false
-                        }
-                    }
-                }
+        filter ?: return
+        if (!filter.validateFilter()) return
 
-                if (it.description != null && it.description!!.isNotEmpty()) {
-                    excludedKeywords.forEach { keyword ->
-                        if (it.description!!.containsLowercase(keyword)) {
-                            return@filter false
-                        }
-                    }
-                }
+        updateState {
+            it.copy(loading = true)
+        }
 
-                true
+        exportCSVLogic.exportToFile(
+            context = context,
+            fileUri = fileUri,
+            exportScope = {
+                allTransactions.value.toOld()
             }
-            .toList()
+        )
+
+        uiThread {
+            onShareUI(fileUri)
+        }
+
+        updateState {
+            it.copy(loading = false)
+        }
     }
 
-    private fun String.containsLowercase(anotherString: String): Boolean {
-        return this.toLowerCaseLocal().contains(anotherString.toLowerCaseLocal())
+    private suspend fun filterTransactions(
+        baseCurrency: String,
+        filter: ReportFilter
+    ) = {
+        ByTypeIn(filter.trnTypes.toNonEmptyList()) and
+                ByAllDate(filter.period) and
+                filterByAccountIn(filter.accounts) and
+                ByCategoryIn(filter.categories.noneCategoryFix().toNonEmptyList())
+    } then queryTransactionsAct then {
+        filterByAmount(baseCurrency = baseCurrency, filter = filter, transList = it)
+    } then {
+        filterByWords(filter, it)
+    } then { allTrans ->
+        val nonPlannedPaymentsTransactions = allTrans.filter { it.time is TrnTime.Actual }
+        val input = CalculateWithTransfersAct.Input(
+            trns = nonPlannedPaymentsTransactions,
+            outputCurrency = baseCurrency,
+            selectedAccounts = filter.accounts
+        )
+        Pair(input, allTrans)
+    } thenInvokeAfter {
+        val (input, allTrans) = it
+        val stats = calculateAct(input)
+
+        Pair(stats, allTrans)
     }
 
-    private fun calculateBalance(incomeExpenseTransferPair: IncomeExpenseTransferPair): BigDecimal {
-        return incomeExpenseTransferPair.income + incomeExpenseTransferPair.transferIncome - incomeExpenseTransferPair.expense - incomeExpenseTransferPair.transferExpense
+    @Suppress("FunctionName")
+    private fun ByAllDate(timePeriod: TimePeriod?): TrnWhere {
```

Comment:
Why not just call it `byDate()` and write docs that it takes both `Actual` and `Due`